src/main/kotlin/cli/Scrape.kt

```diff
@@ -7,25 +7,89 @@ import com.github.ajalt.clikt.parameters.types.int
 import models.SubjectCode
 import models.Term
 import mu.KotlinLogging
-import services.JsonMapper
-import services.scrapeFromCatalog
-import services.scrapeAllFromCatalog
+import scraping.models.Course
+import services.*
 import utils.writeToFileOrStdout
 
 // TODO Change this to package-level protected if that becomes a thing
 /**
- * CLI for querying NYU Albert. Has two subcommands, `catalog` and `section`.
+ * CLI for querying NYU Albert. Has three subcommands, `catalog` and `section` and 'sections'.
  * @author Albert Liu
  */
 internal class Scrape : CliktCommand(name = "scrape") {
 
     init {
-        this.subcommands(Catalog())
+        this.subcommands(Catalog(), Section(), Sections())
     }
 
     override fun run() = Unit
 
     /**
+     * CLI for performing search queries for catalog's section of NYU Albert based registration number.
+     */
+    private class Section() : CliktCommand(name = "section") {
+        private val logger = KotlinLogging.logger("scrape.section")
+        private val term: Term by option("--term").convert {
+            Term.fromId(Integer.parseInt(it))
+        }.required()
+        private val registrationNumber: Int by option(
+                "--registrationNumber",
+                help = "The registration number that you'd use to register for the course."
+        ).int().required()
+        private val file: String? by option("--file")
+        private val batchSize: Int? by option("--batchSize").int()
```

Comment:
I don't think batch size should be an argument if we're only scraping a single section.